crates/router/src/core/payouts.rs

```
@@ -0,0 +1,867 @@
+        }
+    });
+
+    let response = api::PayoutCreateResponse {
+        payout_id: payouts.payout_id.to_owned(),
+        merchant_id: merchant_account.merchant_id.to_owned(),
+        amount: payouts.amount.to_owned(),
+        currency,
+        connector: Some(payout_create.connector.to_owned()),
+        payout_type,
+        billing: address,
+        customer_id,
+        auto_fulfill: payouts.auto_fulfill,
+        email,
+        name,
+        phone,
+        phone_country_code,
+        client_secret: None,
+        return_url: payouts.return_url.to_owned(),
+        business_country: None, // FIXME: Fetch from MCA
+        business_label: None,   // FIXME: Fetch from MCA
+        description: payouts.description.to_owned(),
+        entity_type,
+        recurring: payouts.recurring,
+        metadata: payouts.metadata,
+        status,
+        error_message: payout_create.error_message.to_owned(),
+        error_code: payout_create.error_code,
+    };
+    Ok(services::ApplicationResponse::Json(response))
+}
+
+// DB entries
+pub async fn payout_create_db_entries(
+    state: &AppState,
+    merchant_account: &storage::merchant_account::MerchantAccount,
+    req: &payouts::PayoutCreateRequest,
+    payout_id: &String,
+    connector_name: &api_enums::Connector,
+) -> RouterResult<PayoutData> {
+    let db = &*state.store;
+    let merchant_id = &merchant_account.merchant_id;
+
+    // Get or create customer
+    let customer_details = payments::CustomerDetails {
+        customer_id: req.customer_id.to_owned(),
+        name: req.name.to_owned(),
+        email: req.email.to_owned(),
+        phone: req.phone.to_owned(),
+        phone_country_code: req.phone_country_code.to_owned(),
+    };
+    let customer =
+        helpers::get_or_create_customer_details(state, &customer_details, merchant_account).await?;
+    let customer_id = customer
+        .to_owned()
+        .map_or("".to_string(), |c| c.customer_id);
```

should we throw `missing customer id` error when customer_id is not present?